core/trino-main/src/main/java/io/trino/operator/TaskContext.java

```diff
@@ -514,14 +518,18 @@ public TaskStats getTaskStats()
         Duration fullGcTime = getFullGcTime();
 
         long userMemory = taskMemoryContext.getUserMemory();
+        long systemMemory = taskMemoryContext.getSystemMemory();
 
         synchronized (cumulativeMemoryLock) {
             double sinceLastPeriodMillis = (System.nanoTime() - lastTaskStatCallNanos) / 1_000_000.0;
```

Comment:
I think this `System.nanoTime()` should be saved in a separate variable and lastTaskStatCallNanos should be set to this value near the end of this critical section. Otherwise we might have a few nanos unaccounted for in between.